ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp

```diff
@@ -39,6 +40,111 @@ namespace ngraph
                 }
             } // namespace set_1
 
+            namespace set_12
+            {
+                inline OutputVector dropout(const Node& node)
+                {
+                    auto inputs_size = node.get_ng_inputs().size();
+                    auto outputs_size = node.get_outputs_size();
+                    auto input_data = node.get_ng_inputs().at(0);
+
+                    if (inputs_size == 1 && outputs_size == 1)
+                    {
+                        return {input_data};
+                    }
+                    else
+                    {
+                        // This is work around for hanging optional inputs and optional output
+                        // that are not useful for inference
+                        // Inputs -> Concat -> Split -> Outputs
+                        std::shared_ptr<ngraph::Node> shape_of_data;
+
+                        if (input_data.get_partial_shape().is_static())
+                        {
+                            shape_of_data = default_opset::Constant::create(
+                                ngraph::element::i32,
+                                Shape{input_data.get_partial_shape().rank().get_length()},
+                                input_data.get_partial_shape().to_shape());
+                        }
+                        else
+                        {
+                            shape_of_data = std::make_shared<default_opset::ShapeOf>(input_data);
+                        }
```

Comment:
Consider IIFE here:
```suggestion
                        std::shared_ptr<ngraph::Node> shape_of_data = [&]{
                            if (input_data.get_partial_shape().is_static())
                            {
                                return default_opset::Constant::create(
                                    ngraph::element::i32,
                                    Shape{input_data.get_partial_shape().rank().get_length()},
                                    input_data.get_partial_shape().to_shape());
                            }
                            return std::make_shared<default_opset::ShapeOf>(input_data);
                        }();
```

Reply:
Thank you for your suggestions, but it was only a draft to show and discuss some examples. Resolved with other PR.